### ISSUE TITLE
Non-Hermitian bug coming from eigsolve with complex MPO

### ISSUE THREAD

**Issue by emstoudenmire** [ITensor/ITensors.jl#517]
Forum user rmagaldi reports that the sample code below gives an error (which I can confirm) about 1 out of every 10 times. The error comes from inside KrylovKit, from the call to eigsolve in DMRG. Inside KrylovKit, a test is failing which apparently checks whether the matrix is Hermitian. However, explicitly checking that the "PH" operator is Hermitian inside of DMRG does successfully show that it is. So it's a bit of a mystery what's going on.

Perhaps the non-Hermitianness is very small. Or perhaps there is even a bug in KrylovKit (we are pinned to an older version of it). I'm just listing these as possibilities. I did check that the ProjMPO position! function uses dag correctly when updating the environment tensors.

Sample code:
```
using ITensors

let
  N = 4
  sites = siteinds("Fermion",N,conserve_qns=true)
  t1 = 1.0
  t2p = 1.0+0.1*im
  t2n = conj(t2p)

  ampo = AutoMPO()

  #vertical bonds
  for j in [1,3]
    ampo += -t1,"Cdag",j,"C",j+1
    ampo += -t1,"Cdag",j+1,"C",j
  end

  #horizontal bonds
  for j in [1,2]
    ampo += -t1,"Cdag",j,"C",j+2
    ampo += -t1,"Cdag",j+2,"C",j
  end

  #second-neighbor bonds
  for j=1
    ampo += -t2p,"Cdag",j,"C",j+3
    ampo += -t2n,"Cdag",j+3,"C",j
  end

  for j=2
    ampo += -t2p,"Cdag",j,"C",j+1
    ampo += -t2n,"Cdag",j+1,"C",j
  end

  M = MPO(ampo,sites)

  let
    H = M[1]*M[2]*M[3]*M[4]
    Hc = copy(H)
    Hc = dag(swapprime(H,0,1))
    @show norm(Hc-H)
  end

  println(ampo)
  sweeps = Sweeps(5)
  maxdim!(sweeps,10,20,100,100,200)

  init_state = ["Occ","Emp","Emp","Emp"]

  psi0 = productMPS(sites,init_state)
  #psi0 = randomMPS(sites,init_state,2)

  dmrg(M,psi0,sweeps)

  return
end
```

**Comment by mtfishman** [comment]
@emstoudenmire, using the latest version of ITensors.jl (v0.1.26), which can use KrylovKit v0.5 now, I can't reproduce this error.

Also, I'm a bit confused since in the version above it is using `productMPS`, so I don't see where anything nondeterministic could be happening (my understanding is that Lanczos in KrylovKit should not be nondeterministic since we are always specifying a starting vector).

**Comment by emstoudenmire** [comment]
I just confirmed that I can't reproduce this error anymore, though it was definitely happening every 10 runs or so of the above code with a previous version of ITensors.jl and KrylovKit.jl. But now I believe it's fixed, and the tests I did in the meantime only seemed to suggest that all was well on the ITensor side of things in terms of the DMRG Hamiltonian being Hermitian (even with the older ITensor code). So I suspect it was just a subtle floating point issue, with the floating point roundoff being the source of the apparent randomness.
